bio-demultiplexing-fasta/src/demultiplexed.py

```diff
@@ -28,48 +29,73 @@ def importSample(manifestPath: Path, sequenceType: str, outputDir: Path) -> Path
 
 
 def importMetadata(metadata: CustomSample, outputDir: Path) -> Path:
+    metadata.unzip()
     metadataZipPath = outputDir / "metadata.zip"
+
     with ZipFile(metadataZipPath, "w") as metadataFile:
         metadataFile.write(Path(metadata.path) / "metadata.tsv", "metadata.tsv")
 
     return metadataZipPath
 
 
-def createManifest(samples: List[CustomSample], manifestPath: Path) -> Path:
+def createManifestSingle(samples: List[SequenceSample], manifestPath: Path) -> Path:
     with manifestPath.open("w") as manifestFile:
         csv.writer(manifestFile, delimiter = "\t").writerow(["sample-id", "absolute-filepath"])
 
     for sample in samples:
-        for fastqPath in Path(sample.path).iterdir():
-            if fastqPath.suffix != ".fastq":
-                continue
+        sample.unzip()
+        fastqPath = sample.sequencePath
+
+        with manifestPath.open("a") as manifestFile:
```

Comment:
we should not call open every single time we want to write to a file, especially in a for-loop, this slows things down, let's just open the file once